fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AuthorizerStmtVisitor.java

```diff
@@ -270,13 +297,27 @@ public Void visitUpdateStatement(UpdateStmt statement, ConnectContext context) {
             Authorizer.checkTableAction(context.getCurrentUserIdentity(), context.getCurrentRoleIds(),
                     statement.getTableName(), PrivilegeType.UPDATE);
         } catch (AccessDeniedException e) {
-            AccessDeniedException.reportAccessDenied(statement.getTableName().getCatalog(),
-                    context.getCurrentUserIdentity(), context.getCurrentRoleIds(),
-                    PrivilegeType.UPDATE.name(), ObjectType.TABLE.name(), statement.getTableName().getTbl());
+            Set<String> assignmentColumn = statement.getAssignments().stream()
+                    .map(columnAssignment -> columnAssignment.getColumn()).collect(Collectors.toSet());
+            try {
+                Authorizer.checkColumnsAction(context.getCurrentUserIdentity(), context.getCurrentRoleIds(),
+                        statement.getTableName(), assignmentColumn, PrivilegeType.UPDATE);
+            } catch (AccessDeniedException exception) {
+                AccessDeniedException.reportAccessDenied(statement.getTableName().getCatalog(),
+                        context.getCurrentUserIdentity(), context.getCurrentRoleIds(),
+                        PrivilegeType.UPDATE.name(), ObjectType.COLUMN.name(),
+                        String.join(".", statement.getTableName().getNoClusterString(),
+                                String.join(",", assignmentColumn)));
+            }
         }
         Map<TableName, Relation> allTouchedTables = AnalyzerUtils.collectAllTableAndViewRelations(statement);
-        allTouchedTables.remove(statement.getTableName());
```

Comment:
At the table level, if the UPDATE permission for the same table passes verification, there is no need to check the SELECT permission for that table. However, this logic may not be appropriate in column-level scenarios. Within an UpdateStmt, the columns being updated and those involved in the subquery may not be the same, as in the statement `update t set t.a=1 where t.b=2`, where the user needs both UPDATE privilege for column 'a' and SELECT privilege for column 'b'. It would be better not to remove the target table in this case.